js/ui/PreferencesMenu.js

```diff
@@ -33,8 +33,9 @@ export function __cleanup__() {
 /**
  * Create the preferences menu within the given group.
  * @param {HTMLElement} prefs_group
+ * @param {Function?} [on_change_runtime_channel]
  */
-export function menu_create(prefs_group) {
+export function menu_create(prefs_group, on_change_runtime_channel) {
```

Comment:
I'm not a fan of how this is now passing in a callback, especially given the callback isn't documented in the JSDoc either. I was using a lot of callbacks (see much earlier commits on this) but it got very... unenjoyable lol. I'm not sure if anything I immediately think of would replace this better, just not a massive fan of this. Should be fine though.

Reply:
I think this kinda makes me want events for preferences updates that the control panel can listen for, but that's not something relevant to this PR and is something I'll do later if I've got time.